middleware/id_validation.go

```diff
@@ -0,0 +1,72 @@
+package middleware
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/RedHatInsights/sources-api-go/util"
+	"github.com/labstack/echo/v4"
+)
+
+// IdValidation takes all the parameters which end with "id" and checks for their validity. Returns a bad request if
+// they're not valid.
+func IdValidation(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		for _, param := range c.ParamNames() {
+			if strings.HasSuffix(param, "id") {
+				err := validateId(c, param)
+				if err != nil {
+					return util.NewErrBadRequest(err)
+				}
+			}
+		}
+
+		return next(c)
+	}
+}
+
+// UuidValidation checks if the UUID parameter is valid. Returns a bad request if it isn't.
+func UuidValidation(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		err := validateUuid(c)
+		if err != nil {
+			return util.NewErrBadRequest(err)
+		}
+
+		return next(c)
+	}
+}
+
+// validateId checks if the ID stored in specified context parameter isn't empty, is a valid integer and that it is
+// greater than zero.
+func validateId(c echo.Context, idParamName string) error {
+	idRaw := c.Param(idParamName)
+
+	if idRaw == "" {
```

Comment:
I think this can't happen.

Reply:
You are right, yes. If we send a request to `/sources/` it will just match the collection's route. Thanks!